azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/presentation/fragment/setup/components/JoinCallButtonHolderView.kt

```diff
@@ -46,19 +45,22 @@ internal class JoinCallButtonHolderView : ConstraintLayout {
 
     fun start(
         viewLifecycleOwner: LifecycleOwner,
-        viewModel: JoinCallButtonHolderViewModel,
-        networkManager: NetworkManager
+        viewModel: JoinCallButtonHolderViewModel
     ) {
         this.viewModel = viewModel
-        this.networkManager = networkManager
         setupJoinCallButtonText.text = context.getString(R.string.azure_communication_ui_calling_setup_view_button_join_call)
         joiningCallText.text = context.getString(R.string.azure_communication_ui_calling_setup_view_button_connecting_call)
 
         setupJoinCallButton.setOnClickListener {
-            if (networkManager.isNetworkConnectionAvailable()) {
-                viewModel.launchCallScreen()
-            } else {
+            val networkAvailable = viewModel.isNetworkAvailable()
```

Comment:
all this logic needs to be moved to view model and be covered with unit tests
should be inside launchCallScreen()